packages/core-js/modules/es.regexp.constructor.js

```diff
@@ -98,9 +98,9 @@ var handleNCG = function (string) {
         if (exec(IS_NCG, stringSlice(string, index + 1))) {
           index += 2;
           ncg = true;
+          groupid++;
         }
         result += chr;
-        groupid++;
```

Comment:
It will break cases like
```js
RegExp('foo:(?<foo>\\w+),bar:(\\w+),buz:(?<buz>\\w+)').exec('foo:abc,bar:def,buz:ghi').groups.buz
```
Could you fix it and add a couple of cases like that to the tests?

Reply:
Yes, you are right. Changes like these really break some cases.

I added an additional check to a non-capturing group after '(' character and revert the previous changes.
Also extended regexp constructor test with your example.